frontend/src/components/playgame/MathInput.vue

```diff
@@ -152,21 +158,40 @@ watch(expression, async (newValue) => {
 });
 
 const viewExpression = computed(() => {
-  return expression.value.replace(/-/g, "−").replace(/\*/g, "×").replace(/\//g, "÷");
+  return expression.value
+    .replace(/-/g, "−")
+    .replace(/\*/g, "×")
+    .replace(/\//g, "÷");
 });
 
 const addSymbol = (value: string) => {
-  const last = expression.value.length > 0 ? expression.value[expression.value.length - 1] : "+";
+  const numberCount = (expression.value.match(/[1-9]/g) || []).length;
+  const last =
+    expression.value.length > 0
+      ? expression.value[expression.value.length - 1]
+      : "+";
+
+  // 数字を追加しようとする場合のチェック
+  if (/[1-9]/.test(value)) {
+    // 既に4つの数字が使われている場合は数字の追加を拒否
+    if (numberCount >= 4) {
+      return;
+    }
+  }
+
+  // 演算子を追加しようとする場合のチェック
+  if (/[+\-*/]/.test(value)) {
+    // 4つの数字が使われている場合は新しい演算子の追加を拒否
+    if (numberCount >= 4) {
+      return;
+    }
+  }
```

Comment:
```suggestion
  if (numberCount >= 4) {
    return;
  }
```
実は 1~9 と 記号 との場合分けは不要だったりしませんかね…？ 括弧の入力の際は別の関数が動くので

Reply:
確かにそうかもです！！！
一回コミットしてみて、ほんとに動くかちょっと確認してきます！

Reply:
ちゃんと動いてそうです！！！レビュー有難うございます！